src/main/scala/org/virtuslab/internal/load/construct/YamlDecoder.scala

```diff
@@ -9,29 +8,110 @@ import scala.compiletime._
 import scala.compiletime.summonFrom
 import scala.deriving._
 import scala.util.Try
+import scala.language.implicitConversions
 
 sealed trait YamlDecoder[T]:
-  def construct(node: Node): Either[YamlError, T]
+  def construct(node: Node): Either[ConstructError, T]
 
 object YamlDecoder:
-  def apply[T](pf: PartialFunction[Node, Either[Throwable, T]]): YamlDecoder[T] =
+  def apply[T](pf: PartialFunction[Node, Either[ConstructError, T]]): YamlDecoder[T] =
     new YamlDecoder[T] {
-      override def construct(node: Node): Either[YamlError, T] =
+      override def construct(node: Node): Either[ConstructError, T] =
         if pf.isDefinedAt(node) then
           pf(node) match
             case Right(value)    => Right(value)
             case Left(throwable) => Left(ConstructError(throwable.toString))
         else Left(ConstructError(s"Could't create Construct instance for $node"))
```

Comment:
This apply doesn't work correctly because pf already returs `ConstructError` in the result (`pf: PartialFunction[Node, Either[ConstructError, T]])`)
Maybe it's worth having secondary apply for `Throwable` and get rid of implicit conversion?

Reply:
You are quite right, but it's works because `pf(node)` returned `Either[ConstructError, T]`.

so there was some redundancy here
```
pf(node) match
            case Right(value)    => Right(value)
            case Left(throwable) => Left(ConstructError(throwable.toString))

```
that could be rewritten to `pf(node)`